src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -142,13 +143,33 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
     return false;
 }
 
+double PenInputHandler::inferPressureIfEnabled(PositionInputData const& pos, XojPageView* page) {
+    // Infer pressure
+    if (pos.pressure == Point::NO_PRESSURE && this->inputContext->getSettings()->isPressureGuessingEnabled()) {
+        PositionInputData lastPos = getInputDataRelativeToCurrentPage(page, this->lastEvent);
+
+        double dt = std::min((pos.timestamp - lastPos.timestamp) / 10.0, 2.0);
+        double inverseSpeed = (dt) / std::sqrt(std::pow(pos.x - lastPos.x, 2) + std::pow(pos.y - lastPos.y, 2) + 0.001);
+
+        double newPressure = 3.142 / 2.0 + std::atan(inverseSpeed * 3.14 - 1.3);
```

Comment:
You are using very expensive methods here (pow, sqrt, atan) within a function that is called very very often. It would be good to simplify this calculation to a quick approximation. The result does not need to be perfect as it has no huge influence on the look if the pressure is 0.00000002 or 0.00000003.

Reply:
`std::pow(x,2)` can be replaced by x*x, which is much faster. Unfortunately there is no constexpr version of `<cmath>` functions. Then it wouldn't make any difference.
Also `<cmath>` supports the [`std::hypot`](https://en.cppreference.com/w/cpp/numeric/math/hypot) function, which is optimized for calculating the 2nd norm. If you require precision instead of a numeric approximation, it is still a way faster than a manual `sqrt(pow(x,2) + pow(y,2))` (in this case just use `utl::Point::distance` or `Point::lineLengthTo).

Reply:
`pos` and `lastPos` are of type `PositionInputData`. As such, `utl::Point::distance` does not apply without a conversion.

I am considering making `PositionInputData` extend `utl::Point<double>`...

Reply:
I think extending `util::Point<double>` would be a valid solution.

Reply:
I'm still not sure what to do about `std::atan`...

Reply:
You could use a local object of type utl::Point<double>. Most likely this will completely inlined by the compiler :).

Reply:
Extending `utl::Point<double>` makes the code much cleaner. I think I'll do that for now...